src/NuGet.Services.Metadata.Catalog.Monitoring/ValidationCollectorFactory.cs

```diff
@@ -67,10 +67,10 @@ public ValidationCollectorFactory(ILoggerFactory loggerFactory)
                 messageHandlerFactory,
                 _loggerFactory);
 
-            var endpoints = endpointContexts.Select(e => endpointFactory.Create(e));
+            var endpoints = endpointContexts.Select(e => endpointFactory.Create(e)).ToList();
```

Comment:
Yes, otherwise the list of endpoints is processed multiple times by several resources.